stl/inc/ranges

```diff
@@ -4748,6 +4748,580 @@ namespace ranges {
+                return *_Parent->_Current == _RANGES end(_Parent->_Range) && _Parent->_Remainder != 0;
+            }
+
+            _NODISCARD constexpr range_difference_t<_Vw> _Get_size() const
+                noexcept(noexcept(_Div_ceil(_RANGES end(_Parent->_Range) - *_Parent->_Current, _Parent->_Count))) {
+                const auto _Size = _RANGES end(_Parent->_Range) - *_Parent->_Current;
+                if (_Size < _Parent->_Remainder) {
+                    return _Size == 0 ? 0 : 1;
+                }
+                return _Div_ceil(_Size - _Parent->_Remainder, _Parent->_Count) + 1;
+            }
+
+
+        public:
+            using iterator_concept = input_iterator_tag;
+            using difference_type  = range_difference_t<_Vw>;
+
+            struct value_type : public view_interface<value_type> {
+            private:
+                friend _Outer_iterator;
+                chunk_view* _Parent{};
+
+                class _Inner_iterator {
+                private:
+                    friend value_type;
+                    chunk_view* _Parent{};
+
+                    constexpr explicit _Inner_iterator(chunk_view& _Parent_) noexcept
+                        : _Parent(_STD addressof(_Parent_)) {}
+
+                    _NODISCARD constexpr range_difference_t<_Vw> _Get_remainder() const noexcept {
+                        return _Parent->_Remainder;
+                    }
+
+                    _NODISCARD constexpr range_difference_t<_Vw> _Get_size() const
+                        noexcept(noexcept(_RANGES end(_Parent->_Range) - *_Parent->_Current)) {
+                        return (_STD min)(_Parent->_Remainder, _RANGES end(_Parent->_Range) - *_Parent->_Current);
+                    }
+
+                public:
+                    using iterator_concept = input_iterator_tag;
+                    using difference_type  = range_difference_t<_Vw>;
+                    using value_type       = range_value_t<_Vw>;
+
+                    _Inner_iterator(_Inner_iterator&&) = default;
+                    _Inner_iterator& operator=(_Inner_iterator&&) = default;
+
+                    _NODISCARD constexpr const iterator_t<_Vw>& base() const& noexcept /* strengthened */ {
+                        return *_Parent->_Current;
+                    }
+
+                    _NODISCARD constexpr range_reference_t<_Vw> operator*() const
+                        noexcept(noexcept(**_Parent->_Current)) /* strengthened */ {
+#if _ITERATOR_DEBUG_LEVEL != 0
+                        _STL_VERIFY(*this != default_sentinel, "cannot dereference chunk_view end iterator");
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+                        return **_Parent->_Current;
+                    }
+
+                    constexpr _Inner_iterator& operator++() noexcept(
+                        noexcept(++*_Parent->_Current) && noexcept(_Implicitly_convert_to<bool>(

```

Comment:
```suggestion
                        noexcept(++*_Parent->_Current) && noexcept(static_cast<bool>(
```

`if(...)` does a contextual conversion to `bool` on `...`, which is equivalent to direct initialization ([\[conv.general\]/4]) - i.e., it does an explicit conversion, not an implicit conversion ([\[dcl.init.general\]/16.7]) (`static_cast<bool>(...)` is also a direct initialization ([\[dcl.init.general\]/15.3])).

[\[conv.general\]/4]: https://timsong-cpp.github.io/cppwp/n4868/conv.general#4
[\[dcl.init.general\]/15.3]: https://timsong-cpp.github.io/cppwp/n4868/dcl.init.general#15.3
[\[dcl.init.general\]/16.7]: https://timsong-cpp.github.io/cppwp/n4868/dcl.init.general#16.7

Reply:
Nice catch